### ISSUE TITLE
Try install skeleton v3 with zend-view, asking install/inject Zend\Expressive\Plates\ConfigProvider

### ISSUE THREAD

**Issue by samsonasik** [zendframework/zend-expressive-skeleton#193]
When installer ask for what template to use, I choose `zend-view`:

```
  Which template engine do you want to use?
  [1] Plates
  [2] Twig
  [3] Zend View installs Zend ServiceManager
  [n] None of the above
  Make your selection or type a composer package name and version (n): 3
  - Adding package zendframework/zend-expressive-zendviewrenderer (^2.0.0-dev)
  - Copying config/autoload/templates.global.php
  - Copying templates/error/404.phtml
  - Copying templates/error/error.phtml
  - Copying templates/layout/default.phtml
  - Copying templates/app/home-page.phtml
```

After that, it keep install ` zendframework/zend-expressive-platesrenderer`:
```
  - Installing zendframework/zend-expressive-platesrenderer (dev-release-2.0.0 4ebfbf4): Cloning 4ebfbf4887 from cache

  Please select which config file you wish to inject 'Zend\Expressive\Plates\ConfigProvider' into:
  [0] Do not inject
  [1] config/config.php
  Make your selection (default is 0):

  Remember this option for other packages of the same type? (y/N)y
Installing Zend\Expressive\Plates\ConfigProvider from package zendframework/zend-expressive-platesrenderer
```

I think plates is not needed to be installed when choose another template engine, e.g.: zend-view

Steps to reproduce:

```
$ composer create-project "zendframework/zend-expressive-skeleton:release-3.0.0-dev" expressive-3.0
    1/1:	http://packagist.org/p/provider-latest$b415c2d8d7cbe9e90ea2e81d17bb1b019d99c7d7f963602af197e2a58e495a2b.json
    Finished: success: 1, skipped: 0, failure: 0, total: 1
Installing zendframework/zend-expressive-skeleton (dev-release-3.0.0 e622f55d05b34da2b5ca46b3dffe99dfde1a64ff)
  - Installing zendframework/zend-expressive-skeleton (dev-release-3.0.0 release-3.0.0): Cloning release-3.0.0 from cache
Created project in expressive-3.0
    1/1:	http://packagist.org/p/provider-latest$0a087de73a5065bd1b812bc5c2fefcc9da971790ab1c538a041a81fda2c1a4f5.json
    Finished: success: 1, skipped: 0, failure: 0, total: 1
> ExpressiveInstaller\OptionalPackages::install
Setting up optional packages
Setup data and cache dir
Removing installer development dependencies

  What type of installation would you like?
  [1] Minimal (no default middleware, templates, or assets; configuration only)
  [2] Flat (flat source code structure; default selection)
  [3] Modular (modular source code structure; recommended)
  Make your selection (2): 2
  - Copying src/App/ConfigProvider.php

  Which container do you want to use for dependency injection?
  [1] Aura.Di
  [2] Pimple
  [3] Zend ServiceManager
  [4] Auryn
  [5] Symfony DI Container
  Make your selection or type a composer package name and version (Zend ServiceManager): 3
  - Adding package zendframework/zend-servicemanager (^3.3)
  - Copying config/container.php

  Which router do you want to use?
  [1] Aura.Router
  [2] FastRoute
  [3] Zend Router
  Make your selection or type a composer package name and version (FastRoute): 3
  - Adding package zendframework/zend-expressive-zendrouter (^3.0.0-dev)
  - Copying config/routes.php
  - Copying config/autoload/router.global.php

  Which template engine do you want to use?
  [1] Plates
  [2] Twig
  [3] Zend View installs Zend ServiceManager
  [n] None of the above
  Make your selection or type a composer package name and version (n): 3
  - Adding package zendframework/zend-expressive-zendviewrenderer (^2.0.0-dev)
  - Copying config/autoload/templates.global.php
  - Copying templates/error/404.phtml
  - Copying templates/error/error.phtml
  - Copying templates/layout/default.phtml
  - Copying templates/app/home-page.phtml

  Which error handler do you want to use during development?
  [1] Whoops
  [n] None of the above
  Make your selection or type a composer package name and version (Whoops): 1
  - Adding package filp/whoops (^2.1.12)
  - Copying config/autoload/development.local.php.dist
Remove installer
Removing composer.lock from .gitignore
Removing Expressive installer classes, configuration, tests and docs
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 92 installs, 0 updates, 0 removals
  - Installing zendframework/zend-component-installer (1.1.0): Loading from cache
  - Installing zendframework/zend-expressive-template (dev-release-2.0.0 49a19cf): Cloning 49a19cfe46 from cache
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing http-interop/http-server-handler (1.0.0): Loading from cache
  - Installing http-interop/http-server-middleware (1.0.1): Loading from cache
  - Installing zendframework/zend-escaper (2.5.2): Loading from cache
  - Installing zendframework/zend-stratigility (dev-release-3.0.0 a27cd09): Cloning a27cd09923 from cache
  - Installing fig/http-message-util (1.1.2): Loading from cache
  - Installing zendframework/zend-expressive-router (dev-release-3.0.0 57ebf52): Cloning 57ebf529de from cache
  - Installing zendframework/zend-diactoros (1.6.1): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing zendframework/zend-expressive (dev-release-3.0.0 81a108c): Cloning 81a108ca37 from cache
  - Installing symfony/dependency-injection (v4.0.1): Loading from cache
  - Installing jsoumelidis/zend-sf-di-config (0.1.0): Loading from cache
  - Installing mikey179/vfsstream (v1.6.5): Loading from cache
  - Installing rdlowrey/auryn (v1.4.2): Loading from cache
  - Installing northwoods/container (1.2.0): Loading from cache
  - Installing container-interop/container-interop (1.2.0): Loading from cache
  - Installing aura/di (3.4.0): Loading from cache
  - Installing zendframework/zend-auradi-config (dev-release-1.0.0 459c64b): Cloning 459c64bb43 from cache
  - Installing squizlabs/php_codesniffer (2.9.1): Loading from cache
  - Installing zendframework/zend-coding-standard (1.0.0): Loading from cache
  - Installing psr/log (1.0.2): Loading from cache
  - Installing aura/router (3.1.0): Loading from cache
  - Installing zendframework/zend-expressive-aurarouter (dev-release-3.0.0 7129a05): Cloning 7129a050b4 from cache
  - Installing zendframework/zend-stdlib (3.1.0): Loading from cache
  - Installing nikic/fast-route (v1.2.0): Loading from cache
  - Installing zendframework/zend-expressive-fastroute (dev-release-3.0.0 7987415): Cloning 79874159de from cache
  - Installing zendframework/zend-expressive-helpers (dev-release-5.0.0 0ee725f): Cloning 0ee725f217 from cache
  - Installing league/plates (3.3.0): Loading from cache
  - Installing zendframework/zend-expressive-platesrenderer (dev-release-2.0.0 4ebfbf4): Cloning 4ebfbf4887 from cache

  Please select which config file you wish to inject 'Zend\Expressive\Plates\ConfigProvider' into:
  [0] Do not inject
  [1] config/config.php
  Make your selection (default is 0):

  Remember this option for other packages of the same type? (y/N)y
Installing Zend\Expressive\Plates\ConfigProvider from package zendframework/zend-expressive-platesrenderer
    Package is already registered; skipping
  - Installing symfony/polyfill-mbstring (v1.6.0): Loading from cache
  - Installing twig/twig (v2.4.4): Loading from cache
  - Installing zendframework/zend-expressive-twigrenderer (dev-release-2.0.0 3e2f951): Cloning 3e2f951d18 from cache
  - Installing zendframework/zend-servicemanager (3.3.1): Loading from cache
  - Installing zendframework/zend-validator (2.10.1): Loading from cache
Installing Zend\Validator\ConfigProvider from package zendframework/zend-validator
    Package is already registered; skipping
  - Installing zendframework/zend-uri (2.5.2): Loading from cache
  - Installing zendframework/zend-loader (2.5.1): Loading from cache
  - Installing zendframework/zend-http (2.7.0): Loading from cache
  - Installing zendframework/zend-router (3.0.2): Loading from cache
Installing Zend\Router\ConfigProvider from package zendframework/zend-router
    Package is already registered; skipping
  - Installing zendframework/zend-psr7bridge (1.0.0): Loading from cache
  - Installing zendframework/zend-expressive-zendrouter (dev-release-3.0.0 1234ab0): Cloning 1234ab08bf from cache
  - Installing zendframework/zend-eventmanager (3.2.0): Loading from cache
  - Installing zendframework/zend-view (2.9.0): Loading from cache
  - Installing zendframework/zend-expressive-zendviewrenderer (dev-release-2.0.0 2df57aa): Cloning 2df57aae30 from cache
  - Installing pimple/pimple (v3.2.2): Loading from cache
  - Installing zendframework/zend-pimple-config (dev-release-1.0.0 6c0d886): Cloning 6c0d8866ed from cache
  - Installing zendframework/zend-config-aggregator (1.0.1): Loading from cache
  - Installing symfony/process (v3.4.1): Loading from cache
  - Installing symfony/finder (v3.4.1): Loading from cache
  - Installing symfony/filesystem (v3.4.1): Loading from cache
  - Installing symfony/debug (v4.0.1): Loading from cache
  - Installing symfony/console (v3.4.1): Loading from cache
  - Installing seld/phar-utils (1.0.1): Loading from cache
  - Installing seld/jsonlint (1.6.2): Loading from cache
  - Installing seld/cli-prompt (1.0.3): Loading from cache
  - Installing justinrainbow/json-schema (5.2.6): Loading from cache
  - Installing composer/spdx-licenses (1.1.6): Loading from cache
  - Installing composer/semver (1.4.2): Loading from cache
  - Installing composer/ca-bundle (1.1.0): Loading from cache
  - Installing composer/composer (1.5.5): Loading from cache
  - Installing filp/whoops (2.1.14): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/resource-operations (1.0.0): Loading from cache
  - Installing sebastian/object-reflector (1.1.1): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing sebastian/object-enumerator (3.0.3): Loading from cache
  - Installing sebastian/global-state (2.0.0): Loading from cache
  - Installing sebastian/exporter (3.1.0): Loading from cache
  - Installing sebastian/environment (3.1.0): Loading from cache
  - Installing sebastian/diff (2.0.1): Loading from cache
  - Installing sebastian/comparator (2.1.0): Loading from cache
  - Installing doctrine/instantiator (1.1.0): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/phpunit-mock-objects (5.0.5): Loading from cache
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.5): Loading from cache
  - Installing theseer/tokenizer (1.1.0): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (2.0.2): Loading from cache
  - Installing phpunit/php-code-coverage (5.3.0): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
  - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
  - Installing webmozart/assert (1.2.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.2.0): Loading from cache
  - Installing phpspec/prophecy (1.7.3): Loading from cache
  - Installing phar-io/version (1.0.1): Loading from cache
  - Installing phar-io/manifest (1.0.1): Loading from cache
  - Installing myclabs/deep-copy (1.7.0): Loading from cache
  - Installing phpunit/phpunit (6.5.4): Loading from cache
  - Installing zfcampus/zf-development-mode (3.1.0): Loading from cache
zendframework/zend-expressive suggests installing xtreamwayz/pimple-container-interop (^1.0 to use Pimple for dependency injection)
zendframework/zend-expressive suggests installing zendframework/zend-expressive-tooling (For migration and development tools; require it with the --dev flag)
symfony/dependency-injection suggests installing symfony/yaml ()
symfony/dependency-injection suggests installing symfony/config ()
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
zendframework/zend-expressive-helpers suggests installing mouf/pimple-interop (^1.0 to use Pimple for dependency injection)
zendframework/zend-expressive-platesrenderer suggests installing mouf/pimple-interop (^1.0 to use Pimple for dependency injection)
zendframework/zend-expressive-twigrenderer suggests installing mouf/pimple-interop (^1.0 to use Pimple for dependency injection)
zendframework/zend-servicemanager suggests installing ocramius/proxy-manager (ProxyManager 1.* to handle lazy initialization of services)
zendframework/zend-validator suggests installing zendframework/zend-db (Zend\Db component, required by the (No)RecordExists validator)
zendframework/zend-validator suggests installing zendframework/zend-filter (Zend\Filter component, required by the Digits validator)
zendframework/zend-validator suggests installing zendframework/zend-i18n (Zend\I18n component to allow translation of validation error messages)
zendframework/zend-validator suggests installing zendframework/zend-math (Zend\Math component, required by the Csrf validator)
zendframework/zend-validator suggests installing zendframework/zend-i18n-resources (Translations of validator messages)
zendframework/zend-validator suggests installing zendframework/zend-session (Zend\Session component, ^2.8; required by the Csrf validator)
zendframework/zend-router suggests installing zendframework/zend-i18n (^2.6, if defining translatable HTTP path segments)
zendframework/zend-view suggests installing zendframework/zend-authentication (Zend\Authentication component)
zendframework/zend-view suggests installing zendframework/zend-feed (Zend\Feed component)
zendframework/zend-view suggests installing zendframework/zend-filter (Zend\Filter component)
zendframework/zend-view suggests installing zendframework/zend-i18n (Zend\I18n component)
zendframework/zend-view suggests installing zendframework/zend-json (Zend\Json component)
zendframework/zend-view suggests installing zendframework/zend-mvc (Zend\Mvc component)
zendframework/zend-view suggests installing zendframework/zend-navigation (Zend\Navigation component)
zendframework/zend-view suggests installing zendframework/zend-paginator (Zend\Paginator component)
zendframework/zend-view suggests installing zendframework/zend-permissions-acl (Zend\Permissions\Acl component)
zendframework/zend-config-aggregator suggests installing zendframework/zend-config (Allows loading configuration from XML, INI, YAML, and JSON files)
symfony/console suggests installing symfony/event-dispatcher ()
symfony/console suggests installing symfony/lock ()
filp/whoops suggests installing symfony/var-dumper (Pretty print complex values better with var-dumper available)
filp/whoops suggests installing whoops/soap (Formats errors as SOAP responses)
sebastian/global-state suggests installing ext-uopz (*)
phpunit/phpunit suggests installing phpunit/php-invoker (^1.1)
Writing lock file
Generating autoload files
Do you want to remove the existing VCS (.git, .svn..) history? [Y,n]? Y
> zf-development-mode enable
You are now in development mode.
```

**Comment by samsonasik** [comment]
composer update seems removing the dependencies, it seems fine, the only issue is when I choose zend-view, it should not asking to inject `Zend\Expressive\Plates\ConfigProvider` to config in the fresh install.

**Comment by xtreamwayz** [comment]
Yes, I'm aware of it. It somehow doesn't remove the packages when forcing creating the project from a dev branch. I'll look into it, but I'm not sure if this can be fixed.

**Comment by xtreamwayz** [comment]
Fixed in #195
